xbmc/addons/AddonManager.cpp

```diff
@@ -654,6 +654,12 @@ bool CAddonMgr::GetAddon(const std::string &str, AddonPtr &addon, const TYPE &ty
   return false;
 }
 
+bool CAddonMgr::HasType(const std::string &id, const TYPE &type)
+{
+  AddonPtr addon;
+  return GetAddon(id, addon, type, false);
+}
+
 bool CAddonMgr::FindAddons()
```

Comment:
The FindAddons is also called on `bool CAddonMgr::LoadAddon(const std::string& addonId)`. Maybe add a `bool CAddonMgr::FindAddons(bool initialFind)` or prevent the `m_events.Publish(AddonEvents::Initilized());` from this function complete.

Reply:
Then we should move `m_events.Publish(AddonEvents::Initilized());` to `Init`